flink-python/pyflink/table/table_environment.py

```diff
@@ -756,8 +757,12 @@ def register_function(self, name, function):
         :param function: The python user-defined function to register.
         :type function: pyflink.table.udf.UserDefinedFunctionWrapper
         """
-        self._j_tenv.registerFunction(name, function._judf(self._is_blink_planner,
-                                                           self.get_config()._j_table_config))
+        if isinstance(function, UserDefinedScalarFunctionWrapper):
```

Comment:
We can add a base method in `UserDefinedScalarFunctionWrapper` and call the method here. For example, we can rename the current `_judf` to `_j_user_defined_function`